File: src/lib/file-type/file-type.ts
import { extname } from '@std/path/extname';

export type FileType = 'video' | 'image' | 'sound' | 'directory' | 'unidentified';

export const videoExtensions = ['.mp4', '.mv4', '.m4v'];
export const soundExtensions = ['.mp3'];
export const imageExtensions = ['.png', '.jpg', '.jpeg', '.gif', '.tiff', '.webp', '.avif', '.bmp', '.ico'];

export function isVideoFile(filePath: string): boolean {
    return videoExtensions.includes(extname(filePath).toLowerCase());
}

export function isSoundFile(filePath: string): boolean {
    return soundExtensions.includes(extname(filePath).toLowerCase());
}

export function isImageFile(filePath: string): boolean {
    return imageExtensions.includes(extname(filePath).toLowerCase());
}

File: src/golib/file-type/file_type.go
package filetype

import "strings"

type FileType int

const (
	Video FileType = iota
	Image
	Sound
	Directory
	Undefined
)

var VideoExtensions []string = []string{"mp4", "mv4", "m4v"}
var ImageExtensions []string = []string{"png", "jpg", "jpeg", "gif", "tiff", "webp", "avif", "bmp", "ico"}
var SoundExtensions []string = []string{"mp3"}

func pathEndsInOneOf(path string, exts []string) bool {
	for _, v := range exts {
		if strings.HasSuffix(path, v) {
			return true
		}
	}

	return false
}

func IsVideoFile(filePath string) bool {
	return pathEndsInOneOf(strings.ToLower(filePath), VideoExtensions)
}

func IsImageFile(filePath string) bool {
	return pathEndsInOneOf(strings.ToLower(filePath), ImageExtensions)
}

func IsSoundFile(filePath string) bool {
	return pathEndsInOneOf(strings.ToLower(filePath), SoundExtensions)
}
